pkg/executor/build_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package executor
+
+import (
+	"testing"
+
+	"github.com/GoogleContainerTools/kaniko/pkg/config"
+	"github.com/GoogleContainerTools/kaniko/pkg/dockerfile"
+	"github.com/GoogleContainerTools/kaniko/testutil"
+	"github.com/google/go-containerregistry/pkg/v1"
+)
+
+func Test_reviewConfig(t *testing.T) {
+	tests := []struct {
+		name               string
+		dockerfile         string
+		originalCmd        []string
+		originalEntrypoint []string
+		expectedCmd        []string
+	}{
+		{
+			name: "entrypoint and cmd declared",
+			dockerfile: `
+			FROM scratch
+			CMD ["mycmd"]
+			ENTRYPOINT ["myentrypoint"]`,
+			originalEntrypoint: []string{"myentrypoint"},
```

Comment:
by originalCmd, do you mean command from previous stage?

Reply:
use different values for dockerfile.CMD and dockerfile.ENTRYPOINT.

Reply:
by originalCmd, I mean the original value of Cmd in the config file (it should match the value specified in the Dockerfile)

The function reviews the Dockerfile and then clears the Cmd field if it isn't declared in the Dockerfile.